portfolio/src/main/webapp/index.html

```diff
@@ -108,13 +108,23 @@ <h3 id="contact_me">Contact Me</h3>
                placeholder="example@gmail.com"><br>
         <input class="button" type="submit" value="Submit">
       </form>
+      <div id="is-latest-input-dangerous"></div>
       <p><b>Comment History</b></p>
-      <label for="maxCommentsToDisplay">Maximum number of entries to show in comment history:</label>
-      <input type="number" id="maxCommentsToDisplay" name="maxCommentsToDisplay"
-             onchange="addComments()" min="0" max="" value="">
-      <p>(The latest comments are displayed on top.)</p>
       <ul id="comment-container"></ul>
-      <button onclick="addComments()">Refresh</button>
+      <div id="pagination">
+        <label for="max-comments-to-display">Maximum number of entries to show on one page of the comment history:</label>
+        <input type="number" id="max-comments-to-display" name="max-comments-to-display" onchange="addComments(1)"
+               min="0" max="" value="">
+        <p>Currently on page <b id="current-page-id"></b>. (The latest comments are displayed on top.)</p>
+        <button id="first" onclick="addComments(this.id)">&lt;&lt;</button>
+        <button id="prev" onclick="addComments(this.id)">&lt;</button>
+        <button id="next" onclick="addComments(this.id)">&gt;</button>
+        <button id="last" onclick="addComments(this.id)">&gt;&gt;</button>
+        <label for="go-to-page-id">Go to page </label>
+        <input type="number" id="go-to-page-id" name="go-to-page-id" onchange="addComments(this.value)"
+               min="0" max="" value="1">
```

Comment:
I would make the min here and for the max-comments-to-display 1 since there is no page "0" and setting max comments to 0 causes weirdness when trying to go to the last page.

Reply:
Got it, I will fix this

Reply:
This is fixed.